cache/disk/disk.go

```diff
@@ -90,17 +107,35 @@ func New(dir string, maxSizeBytes int64, proxy cache.Proxy) (*Cache, error) {
 	// This function is only called while the lock is held
 	// by the current goroutine.
 	onEvict := func(key Key, value lruItem) {
-		f := filepath.Join(dir, key.(string))
+		ks := key.(string)
+		hash := ks[len(ks)-sha256.Size*2:]
+		var kind cache.EntryKind = cache.AC
+		if strings.HasPrefix(ks, "cas") {
+			kind = cache.CAS
```

Comment:
This needs to be updated to handle uncompressed .v1 files.

I should also add back some eviction tests with uncompressed storage mode, since the size is predictable again.

Reply:
The proxy interface also needs fixing.